src/plugins/data/server/search/strategies/eql_search/response_utils.ts

```diff
@@ -24,3 +25,40 @@ export function toEqlKibanaSearchResponse(
     isRunning: response.body.is_running,
   };
 }
+
+export function errorToEqlKibanaSearchResponse(
+  response: EqlResponseError
+): EqlSearchStrategyResponse {
+  return {
+    id: undefined,
+    rawResponse: response.meta,
+    isPartial: false,
+    isRunning: false,
+  };
+}
+
+const PARSING_ERROR_TYPE = 'parsing_exception';
+const VERIFICATION_ERROR_TYPE = 'verification_exception';
+const MAPPING_ERROR_TYPE = 'mapping_exception';
+
+interface ErrorCause {
+  type: string;
+  reason: string;
+}
+
+interface EqlErrorBody {
+  error: ErrorCause & { root_cause: ErrorCause[] };
+}
+
+export interface EqlResponseError extends errors.ResponseError {
+  meta: TransportResult<EqlErrorBody>;
+}
+
+const isValidationErrorType = (type: unknown): boolean =>
+  type === PARSING_ERROR_TYPE || type === VERIFICATION_ERROR_TYPE || type === MAPPING_ERROR_TYPE;
+
+export const isEqlResponseError = (response: unknown): response is EqlResponseError =>
```

Comment:
A few of these methods are unique to this situation, but most of the code here is identical to that in [common security_solution code](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/search_strategy/eql/validation/helpers.ts) (which also exists in [timelines](https://github.com/elastic/kibana/blob/main/x-pack/plugins/timelines/common/search_strategy/eql/validation/helpers.ts) 😭 ). Is this the right time for a package?